integrations/access/pagerduty/client.go

```diff
@@ -136,11 +136,18 @@ func onAfterPagerDutyResponse(sink common.StatusSink) resty.ResponseMiddleware {
 		}
 
 		if resp.IsError() {
-			result := resp.Error()
-			if result, ok := result.(*ErrorResult); ok {
-				return trace.Errorf("http error code=%v, err_code=%v, message=%v, errors=[%v]", resp.StatusCode(), result.Code, result.Message, strings.Join(result.Errors, ", "))
+			var details string
+			switch result := resp.Error().(type) {
+			case *ErrorResult:
```

Comment:
nit: Can you use `errors.As` to prevent the type check failing when the error is wrapped?

Reply:
Ah, I see the confusion here: `ErrorResult` is not a Go `error` - it describes the structured PagerDuty error reponse body.

I might try and rename stuff, or at least comment it a bit, to try and explain what's going on.